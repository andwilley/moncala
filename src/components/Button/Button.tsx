import React from 'react';

interface IButtonProps {
  disabled?: boolean;
  hasBorder?: boolean;
  name?: string;
  className?: string;
  onClick?: (e: React.MouseEvent) => void;
  type?: 'button' | 'submit' | 'reset';
}

const Button: React.FC<IButtonProps> = ({
  children,
  disabled = false,
  hasBorder = false,
  className = '',
  name,
  onClick,
  type = 'button'
}) => (
  <button
    className={`${
      hasBorder ? 'ba' : 'bn'
    } bg-transparent pr2 dim f6 gray ba b--moon-gray pointer br2 ${className}`}
    disabled={disabled}
    name={name}
    onClick={onClick}
    type={type}
  >
    {children}
  </button>
);

export default Button;
