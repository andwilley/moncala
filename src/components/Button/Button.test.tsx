import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

describe('<Button> test a basic render output', () => {
  it('should render the notepad', () => {
    const tree = renderer.create(<Button>Child</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
