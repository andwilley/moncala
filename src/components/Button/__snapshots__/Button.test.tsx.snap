// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> test a basic render output should render the notepad 1`] = `
<button
  className="bn bg-transparent pr2 dim f6 gray ba b--moon-gray pointer br2 "
  disabled={false}
  type="button"
>
  Child
</button>
`;
